include/seqan3/io/alignment_file/output.hpp

```diff
@@ -0,0 +1,940 @@
+            detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::QUAL)>(t)),
+            detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::SEQ_QUAL)>(t)),
+            detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::MATE)>(t)),
+            detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::TAGS)>(t)),
+            detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::EVALUE)>(t)),
+            detail::range_wrap_ignore(detail::get_or_ignore<selected_field_ids::index_of(field::BIT_SCORE)>(t)));
+    }
+    //!\}
+
+    //!\brief The options are public and its members can be set directly.
+    alignment_file_out_options options;
+
+    /*!\cond DEV
+     * \brief Expose a reference to the underlying stream object. [public, but not documented as part of the API]
+     */
+    stream_type & get_stream()
+    {
+        return stream;
+    }
+    //!\endcond
+
+protected:
+    //!\privatesection
+    //!\brief Path of the file that the stream operates on.
+    std::string file_name; // TODO why not a filesystem::path?
+
+    //!\brief The stream we are writing to.
+    stream_type stream;
+
+    //!\brief Type of the format, an std::variant over the `valid_formats`.
+    using format_type = detail::transfer_template_args_onto_t<valid_formats, std::variant>;
+    //!\brief The actual std::variant holding a pointer to the detected/selected format.
+    format_type format;
+
+    //!\brief Write record to format.
+    template <typename header_type,
+              typename seq_type,
+              typename id_type,
+              typename offset_type,
+              typename ref_seq_type,
+              typename ref_id_type,
+              typename ref_offset_type,
+              typename align_type,
+              typename flag_type,
+              typename mapq_type,
+              typename qual_type,
+              typename seq_qual_type,
+              typename mate_type,
+              typename tags_type,
+              typename evalue_type,
+              typename bit_score_type>
+    void write_record(header_type      && header,
+                      seq_type         && seq,
+                      id_type          && id,
+                      offset_type      && offset,
+                      ref_seq_type     && ref_seq,
+                      ref_id_type      && ref_id,
+                      ref_offset_type  && ref_offset,
+                      align_type       && align,
+                      flag_type        && flag,
+                      mapq_type        && mapq,
+                      qual_type        && qual,
+                      seq_qual_type    && seq_qual,
+                      mate_type        && mate,
+                      tags_type        && tags,
+                      evalue_type      && evalue,
+                      bit_score_type   && bit_score)
```

Comment:
could be simplified by parameter packs.

Reply:
Yes but no because some of the types/values are treated special and would need to passed as single paramters (namely header, seq, qual, seq_qual). I could put all of those in the beginning and pack the rest ?